android-core/src/main/java/com/mparticle/internal/ConfigManager.java

```diff
@@ -1276,6 +1283,23 @@ public ConsentState getConsentState(long mpid) {
         return ConsentState.withConsentState(serializedConsent).build();
     }
 
+    public boolean isDirectUrlRoutingEnabled() {
+        return directUrlRouting;
+    }
+
+    public String getPodPrefix() {
+        String prefix = "us1";
+        try {
+            String[] prefixFromApi = getApiKey().split("-");
+            if (prefixFromApi.length > 1) {
+                prefix = getApiKey().split("-")[0];
+            }
+        } catch (Exception e) {
+            prefix = "us1";
+        }
+        return prefix;
```

Comment:
I don't think that a try/catch is appropriate here.  try/catch should be used to code safely in case an unexpected exception is thrown.  In this case, we know there will or will not be a prefix. 
 It's not that the scenario without a prefix is an exception/error.  I think this should just be the if/else that you have on line 1294, but the `else` shoudl be defining the prefix as us1, as opposed to an error being caught.

Reply:
https://github.com/mParticle/mparticle-web-sdk/pull/801/files#diff-2717c7fb31c2070dd96f07394997809bc9be7f3fff03ab638db696cc480f39b6R491-R498

Reply:
Account for the length being 0 in the if else clause.